src/main/java/land/oras/auth/FileStoreAuthenticationProvider.java

```diff
@@ -0,0 +1,81 @@
+package land.oras.auth;
+
+import land.oras.credentials.FileStore;
+import land.oras.credentials.FileStore.Credential;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+
+/**
+ * FileStoreAuthenticationProvider is an implementation of the AuthProvider interface.
+ * It retrieves credentials from a FileStore and generates a Basic Authentication header.
+ */
+public class FileStoreAuthenticationProvider implements AuthProvider {
```

Comment:
We can delegate to https://github.com/oras-project/oras-java/blob/16e62f784d680bd101a1bf87b2756f6731f3aa7e/src/main/java/land/oras/auth/UsernamePasswordProvider.java

Instead of storaing username/password there 

example (to improve)

```java
public FileStoreAuthenticationProvider(FileStore fileStore, String serverAddress) {
  this.fileStore = fileStore;
  this.serverAddress = serverAddress;
  Credential credential = fileStore.get(serverAddress);
  this.usernamePasswordAuthProvider = new UsernamePasswordProvider(credential.getUsername(), credential.getPassword());

public String getAuthHeader() {
   return usernamePasswordAuthProvider.getAuthHeader()
}
```

Reply:
Done @jonesbusy